### PR TITLE
Add some setup documentation

### PR DESCRIPTION
**Pull request by Peque**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -91,7 +92,7 @@ static void after_moving(void)
  *
  * @param[in] run Whether the robot should be running.
  */
-static void movement_phase(bool do_run)
+static void configure_speed(bool do_run)
```
Comment (cua-cua) [inline src/main.c]: Currently, this function is not only configuring the speed, but it is also moving the robot. I think that the name and description is unclear. It is hard to follow the code (comparing with previous exploration-movement phase functions)

File: src/main.c [hunk 2]
```diff
@@ -111,10 +114,11 @@ static void movement_phase(bool do_run)
 }
 
 /**
- * @brief Configure the goal for the search phase.
+ * @brief Let the user configure the goal for the search phase.
```
Comment (cua-cua) [inline src/main.c]: Adding configure_speed(false), this function it is also moving the robot on exploration phase. The same that for previous function, I think that the name and comments are unclear and is hard to follow the code.

File: src/main.c [hunk 3]
```diff
@@ -124,44 +128,40 @@ static void configure_goal(void)
...
+}
+
+/**
+ * @brief Start the robot configuration process for an exploration/run.
+ */
+static void configure_start(void)
+{
+	if (maze_is_saved()) {
+		led_bluepill_on();
+		configure_reuse();
```
Comment (cua-cua) [inline src/main.c]: The configure_reuse function includes configure_goal, that includes configure_speed, it is hard to understand the paths. Could you group on a different manner?

File: docs/source/setup.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
+*****
+Setup
+*****
+
+
+User interface
```
[the remark this answers is not in the document]
Comment (Peque) [reply docs/source/setup.rst]: Could you propose a better structure?

File: src/main.c (hunk 1 above)
Comment (Peque) [reply src/main.c]: [Reply on a different comment]

File: src/main.c (hunk 2 above)
Comment (Peque) [reply src/main.c]: [Reply on a different comment]

File: src/main.c (hunk 3 above)
Comment (Peque) [reply src/main.c]: Replying to this and the previous comments:

I disagree. I think this implementation is easier to follow (although different, and that may be the cause why it feels harder to follow at first, since you are probably used to the previous code).

There is a huge improvement with this PR: this setup/configuration code is documented with a graph. If you have not tried to generate the documentation locally, here is the result:

graphviz-927c595d49bbfda9cd978ebf7b9de8e960ed5ff6

This is objectively simpler than the previous implementation.

If you can suggest a simpler configuration procedure, I will change it of course. But cannot think of a better way right now. :innocent:

Comment (cua-cua) [reply src/main.c]: I like a lot the configuration and the user state machine :), the comments were related to the code, or how that is implemented (functions names and dependencies)

File: docs/source/setup.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (cua-cua) [reply docs/source/setup.rst]: Ok, maybe this part could be "User interface" and the IR and encoders calibration "Calibration"?
